products/data_modeling/backend/models/edge.py

```diff
@@ -68,14 +68,12 @@ class Meta:
         ]
 
     def save(self, *args, **kwargs):
-        self._detect_cycles()
-        self._detect_dag_mismatch()
-        super().save(*args, **kwargs)
+        with transaction.atomic():
+            self._detect_cycles()
+            self._detect_dag_mismatch()
+            super().save(*args, **kwargs)
 
     def _detect_cycles(self):
-        # acquire advisory lock to prevent race conditions
-        with connection.cursor() as cursor:
-            cursor.execute("SELECT pg_advisory_xact_lock(%s, hashtext(%s))", [self.team_id, self.dag_id])
```

Comment:
We dont wanna fully delete this, right? We still want this lock, but it just needed to be inside the transaction